include/eve/bundle.hpp

```diff
@@ -0,0 +1,77 @@
+//==================================================================================================
+/**
+  EVE - Expressive Vector Engine
+  Copyright : EVE Contributors & Maintainers
+  SPDX-License-Identifier: MIT
+**/
+//==================================================================================================
+#pragma once
+
+#include <eve/detail/kumi.hpp>
+#include <eve/traits/cardinal.hpp>
+#include <eve/traits/element_type.hpp>
+#include <eve/concept/vectorized.hpp>
+
+namespace eve
+{
+  template<simd_value... Members> struct bundle;
+
+  template<simd_value Member, simd_value... Members>
+  requires( (Member::size() == Members::size()) && ... && true )
+  struct bundle<Member, Members...>
+  {
+    using is_product_type = void;
+
+    bundle() = default;
+
+    bundle(Member const& m,Members const&... ms)    : contents{m,ms...} {}
+    bundle(kumi::tuple<Member,Members...> const& t) : contents{t} {}
+
+    static constexpr auto size() noexcept { return 1 + sizeof...(Members); }
+
+    template<std::size_t I>
+    friend auto const& get(bundle const& s) noexcept { return kumi::get<I>(s.contents); }
+
+    template<std::size_t I>
+    friend auto & get(bundle& s) noexcept { return kumi::get<I>(s.contents); }
+
+    friend std::ostream &operator<<(std::ostream &os, bundle const &p)
+    {
+      return os << p.contents;
+    }
+
+    kumi::tuple<Member, Members...> contents;
+  };
+
+  template<simd_value... Members> bundle(kumi::tuple<Members...> const&)  -> bundle<Members...>;
+  template<simd_value... Members> bundle(Members const&...)               -> bundle<Members...>;
+
+  template<typename Function, kumi::product_type... Bundles>
+  decltype(auto) map(Function&& f, Bundles const&... bs)
+  {
+    return bundle{kumi::map(f, bs...)};
+  }
```

Comment:
Let's make these friends?

Can we still do `kumi` operations directly to get a `kumi::tuple`? We'll need it.

Reply:
you can yes. 
friends is obviosu yes